### PR TITLE
Add Parasoft dotTEST code-scanning workflow

### PR DESCRIPTION
**Pull request by tobyash86**
- partnership pre-requisite is met
- parasoft-dottest.yml workflow script added under code-scanning folder
- matching `code-scanning/properties/parasoft-dottest.properties.json` with all the required properties set, with a single exception: used `creator` instead of `organization` based on other code-scanning integrations
- the workflow has recommended triggers defined
- it uses only GitHub or Parasoft actions
- Parasoft action is referenced using commit hash
- mandatory comment included at the top of the workflow file

Please review.

### CODE REVIEW

File: code-scanning/parasoft-dottest.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+# This workflow runs Parasoft dotTEST to analyze code
+# and display results with Github code scanning alerts.
```
Comment (imjohnbo) [inline code-scanning/parasoft-dottest.yml:7]:
```suggestion
# and display results with GitHub code scanning alerts.
```

File: code-scanning/parasoft-dottest.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    - cron: $cron-weekly
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel.
+jobs:
+  # This workflow is made up of one job "run-dottest-action".
+  run-dottest-action:
+    # Specifies the name of the job.
+    name: Run code analysis with dotTEST
+    # Specifies the type of runner that the job will run on.
+    runs-on: self-hosted
```
Comment (imjohnbo) [inline code-scanning/parasoft-dottest.yml:30]:
```suggestion
    runs-on: ubuntu-20.04
```

**Review by imjohnbo — CHANGES_REQUESTED** [review]
Thanks for the pull request, @tobyash86! A couple of small changes requested.

File: code-scanning/properties/parasoft-dottest.properties.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+  
+{
+    "name": "Parasoft dotTEST Code Analysis",
+    "creator": "Parasoft",
+    "description": "Code analysis with Parasoft dotTEST for C# and VB.NET to ensure code quality and compliance with CWE, OWASP, and more.",
+    "iconName": "icons/parasoft.svg",
```
Comment (imjohnbo) [inline code-scanning/properties/parasoft-dottest.properties.json:6]:
```suggestion
    "iconName": "parasoft",
```

**Review by imjohnbo — CHANGES_REQUESTED** [review]

File: code-scanning/parasoft-dottest.yml (hunk 2 above)
Comment (tobyash86) [reply code-scanning/parasoft-dottest.yml:30]: Current licensing mechanism does not allow to run the analysis on GitHub hosted runners. Is it a blocker? We are working to resolve this. I expect this to be working in the second half of the year.

**Comment by tobyash86** [comment]
Thanks @imjohnbo for helping me out with this. One issue is still opened, please refer to my comment and let me know if this is acceptable approach for now.

File: code-scanning/parasoft-dottest.yml (hunk 2 above)
Comment (imjohnbo) [reply code-scanning/parasoft-dottest.yml:30]: I'm not sure this is the user experience we want to promote in this starter workflow. @josepalafox will follow up with you by email 🙇.

**Comment by imjohnbo** [comment]
> Probably this is not the best place to ask this question, but maybe you could help me with this... I need to set custom icon for Parasoft action. How can I do that? Can I do it by myself or do I need GitHub team assistance? In the latter case, who should I ask for help?

Good question – I believe this is a benefit of the [GitHub Technology Partner Program](https://partner.github.com/technology-partners) and will be arranged through the application process.

**Comment by Ruby-star-cloud** [comment]
Open

File: code-scanning/parasoft-dottest.yml (hunk 2 above)
Comment (tobyash86) [reply code-scanning/parasoft-dottest.yml:30]: @imjohnbo @josepalafox, based on e-mail conversation, is it still a blocker or can we proceed and accept this approach?

**Pull request closed without merging**